src/stdlib.sol

```diff
@@ -65,19 +67,53 @@ abstract contract stdCheats {
         vm_std_cheats.startPrank(who, origin);
     }
 
-    // Allows you to set the balance of an account for a majority of tokens
-    // Be careful not to break something!
+    // DEPRECATED: Use `allot` instead
     function tip(address token, address to, uint256 give) public {
+        emit WARNING_Deprecated("The `tip` stdcheat has been deprecated. Use `allot` instead.");
+        std_store_std_cheats
+            .target(token)
+            .sig(0x70a08231)
+            .with_key(to)
+            .checked_write(give);
+    }
+
+    // Set the balance of an account for any ERC20 token
+    // Use the alternative signature to update `totalSupply`
+    function allot(address token, address to, uint256 give) public {
```

Comment:
Thoughts on just calling this `deal` like the standard cheatcode for setting ETH balance? Seems unnecessary to have two different method names, plus you can special case the `0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE` address to fallback to `vm.deal`, that way all ETH/token balance setting has the same interface

Reply:
Part of me likes it, part me feels like it clutters the interface because you can accomplish the same by only setting balances directly, e.g. `deal(dai, address(this), dai.balanceOf(address(this)) + 10000e18)`. I think I lean towards keeping the interface simpler and have `deal` just set directly the balance

Reply:
i like moving to `deal` and i like having token at the front personally

Reply:
i'm working on this now

Reply:
how do you guys think we should handle breaking changes like this?

Reply:
Right now we have no tags/releases, so my suggestion would be to tag the latest master (d880879) as `v0.1.0`, then once this is merged tag it as `v0.2.0`. Going forwards we'd then bump the patch version on each merge to master (or at least each merge worthy of a release), and bump the minor version on breaking changes like this one. The `v1.0.0` release of forge-std can be roughly synced with foundry's 1.0.0 release

This also helps forge-std prepare for https://github.com/gakonst/foundry/issues/401

Reply:
Sounds good.

1. Release `v0.1.0` now
2. Merge this breaking change and release `v0.2.0`
3. Add new feature and release `v0.2.1`
4. Major change something and tag `v0.2.2`
5. Add new feature and release `v0.2.3`
6. Minor change something
7. \* PR with breaking change(s) \*
8. Release `v0.2.4` with work so far, if not released
9. Merge the breaking change(s) and release `v0.3.0`

Would you change something here?

Release notes:

1. new features
2. breaking changes
3. fixes
4. anything else

Reply:
These are the release notes for `v0.2.0`

## New features

- New stdcheat: **`deal`** lets you set the ETH or ERC20 token balance of an account from the same interface (no need to use Hevm's `deal` for ETH and a separte stdcheat for ERC20 tokens anymore).
- The `deal` stdcheat (previously `tip`) can now optionally adjust the total supply of an ERC20 token after changes have been made to an account's balance.

## Breaking changes

- The `tip` stdcheat has been deprecated. Use the new `deal` stdcheat instead.
- The `lowLevelError` has been deprecated. Use Hevm's `expectRevert` without any arguments instead.

## Fixes

- Fix a test that was causing the `Unused local variable` compiler warning.